node/src/testing/network.rs

```diff
@@ -28,9 +30,20 @@ use crate::{
 pub(crate) type Nodes<R> = HashMap<NodeId, Runner<ConditionCheckReactor<R>>>;
 
 /// A reactor with networking functionality.
+///
+/// Test reactors implementing this SHOULD implement at least the `node_id` function if they have
+/// proper networking functionality.
 pub(crate) trait NetworkedReactor: Sized {
     /// Returns the node ID assigned to this specific reactor instance.
-    fn node_id(&self) -> NodeId;
+    ///
+    /// The default implementation generates a pseudo-id base on its memory address.
+    fn node_id(&self) -> NodeId {
+        #[allow(trivial_casts)]
+        let addr = self as *const _ as usize;
+        let mut raw: [u8; KeyFingerprint::LENGTH] = [0; KeyFingerprint::LENGTH];
+        raw[0..(mem::size_of::<usize>())].copy_from_slice(&addr.to_be_bytes());
+        NodeId::from(KeyFingerprint::from(raw))
+    }
```

Comment:
Just for my own understanding - how is this change related to the task? Did you stumble across this bit when testing the queue dump functionality?

Reply:
The test reactor for queue dumping has no networking component, but to use the `Network` testing structure it expects a `NetworkedReactor` implementation.

This is a bit of a misnomer, it does not need an actual network, but rather a unique ID per node. So instead of cramming a networking implementation into my tiny test-reactor, I instead decided to provide a unique ID based on something other than a random network ID, in this case the memory address of the instantiated node.

As it turns out, this makes a reasonable default implementation.